assignment-02/queries.sql

```diff
@@ -0,0 +1,558 @@
+        and employee.LastName = 'Dodsworth';
+                
+-- 5
+-- How many orders have been placed in total by Robert King
+select 
+    count(*)
+from
+    orders
+        left join
+    employee ON employee.EmployeeID = orders.EmployeeID
+where
+    employee.FirstName = 'Robert'
+        and employee.LastName = 'King';
+        
+-- 6
+-- How many products have been ordered by Robert King between 15th August,1996 and 15th August,1997
+select 
+    count(*)
+from
+    orders
+        left join
+    employee ON employee.EmployeeID = orders.EmployeeID
+where
+    employee.FirstName = 'Robert'
+        and employee.LastName = 'King'
+        and orders.OrderDate between '1996-08-15' and '1997-08-15';
+        
+-- 7
+/* I want to make a phone call to the employees to wish them on the occasion of Christmas who placed
+orders between 13th of January,1997 and 16th of April,1997. I want the EmployeeID, Employee Full Name,
+HomePhone Number. */
+select distinct
+    (employee.EmployeeID),
+    concat_ws(' ',
+            employee.FirstName,
+            employee.LastName) as EmployeeFullName,
+    employee.HomePhone
+from
+    orders
+        left join
+    employee ON employee.EmployeeID = orders.EmployeeID
+where
+    orders.OrderDate between '1997-01-13' and '1997-04-16';
+    
+-- 8
+-- Which product received the most orders. Get the product's ID and Name and number of orders it received.
+with product_orders as (
+select 
+    orderdetails.ProductID, products.ProductName, count(orderdetails.ProductID) as number_of_orders
+from
+    orderdetails
+        left join
+    products on orderdetails.ProductID = products.ProductID group by orderdetails.ProductID) 
+select * from product_orders 
+where number_of_orders=(select max(number_of_orders) from product_orders);
+
+-- 9
+-- Which are the least shipped products. List only the top 5 from your list.
+select 
+    orderdetails.ProductID,
+    products.ProductName,
+    count(orderdetails.ProductID) as number_of_orders
+from
+    orderdetails
+        left join
+    products ON orderdetails.ProductID = products.ProductID
+group by orderdetails.ProductID
+order by number_of_orders
+limit 5;
+
+-- 10
+-- What is the total price that is to be paid by Laura Callahan for the order placed on 13th of January,1997
+select 
+    sum((orderdetails.Quantity * orderdetails.UnitPrice)) as total_price
+from
+    orderdetails
+        join
+    orders ON orders.OrderID = orderdetails.OrderID
+where
+    orders.EmployeeID = (select 
+            EmployeeID
+        from
+            employee
+        where
+            FirstName = 'Laura'
+                and LastName = 'Callahan')
+        and orders.OrderDate = '1997-01-13';
+    
+-- 11
+/* How many number of unique employees placed orders for Gorgonzola Telino or Gnocchi di nonna Alice or
+Raclette Courdavault or Camembert Pierrot in the month January,1997 */
+select 
+    count(distinct orders.EmployeeID) as number_of_employees
```

Comment:
in where clause, instead of between - just compare only year and month

Reply:
comparing year and month instead of using between